src/Core/Checkout/Cart/Rule/LineItemListPriceRule.php

```diff
@@ -61,6 +62,10 @@ public function getConstraints(): array
         ];
 
         if ($this->operator === self::OPERATOR_EMPTY) {
+            $constraints['amount'] = new EqualTo([
+                'value' => 0,
+            ]);
```

Comment:
Generally speaking it is correct that the constraints should not include `amount` if the `empty` operator is used. If the rule is persisted via the administration `amount` should be unset if the operator is switched to `empty`. The real question is why the rule is persisted when persisting an order and why an amount is assigned when it previously wasn't. I feel like if that is the case, that's something that should be fixed at that point instead.